core/deployment/src/main/java/io/quarkus/deployment/ExtensionLoader.java

```diff
@@ -183,9 +200,13 @@ private static boolean isRecorder(AnnotatedElement element) {
                 reader.getBuildTimePatternMap());
         final DefaultValuesConfigurationSource ds2 = new DefaultValuesConfigurationSource(
                 reader.getBuildTimeRunTimePatternMap());
+        final PropertiesConfigSource platformConfigSource = new PropertiesConfigSource(platformProperties, "Quarkus platform",
+                // Our default value configuration source is using an ordinal of Integer.MIN_VALUE
+                // (see io.quarkus.deployment.configuration.DefaultValuesConfigurationSource)
+                Integer.MIN_VALUE + 1000);
```

Comment:
I was thinking about this further and I had a concern.  The platform configuration source most likely should be based on a `io.smallrye.config.KeyMapBackedConfigSource` which can have a default value that matches a pattern.  This would allow configuration keys which have map key components to have a default value.

Reply:
Should I be looking into creating values as nesed KeyMaps?

Reply:
Right, IIRC it would look something like this to add some key under `quarkus.foo.bar.*.baz`:

```java
keyMap.findOrCreate("quarkus", "foo", "bar").getOrCreateAny().findOrCreate("baz").putRootValue("the-default-value");
```

Reply:
Is there some kind of utility for that I should be splitting the key "manually"?

Reply:
There's no canonical input format for key _patterns_ (just the conventional style I used above), so there isn't really an answer to that question.  Ideally we could avoid having any kind of flat map (e.g. properties) as an input source.  Something hierarchical (e.g. YAML) could sidestep the problem.  Otherwise, yes, you'll have to devise an unambiguous syntax and a splitter for that syntax.